cloudant-spring-boot-starter/build.gradle

```diff
@@ -26,9 +26,52 @@ buildscript {
 }
 
 dependencies {
-  compile group: 'org.springframework.boot', name: 'spring-boot', version: springBootVersion
-  compile group: 'org.springframework.boot', name: 'spring-boot-autoconfigure', version: springBootVersion
-  optional group: 'org.springframework.boot', name: 'spring-boot-configuration-processor', version: springBootVersion
+  compile group: 'org.springframework.boot', name: 'spring-boot', version: compileWithBootVersion
+  compile group: 'org.springframework.boot', name: 'spring-boot-autoconfigure', version: compileWithBootVersion
+  optional group: 'org.springframework.boot', name: 'spring-boot-configuration-processor', version: compileWithBootVersion
+}
+
+configurations {
```

Comment:
I think all of this can be replaced by two `testImplementation` dependencies above:
```groovy
    testImplementation group: 'org.springframework.boot', name: 'spring-boot', version: testWithBootVersion
    testImplementation group: 'org.springframework.boot', name: 'spring-boot-autoconfigure', version: testWithBootVersion
```

Reply:
Not entirely sure what you are highlighting with the comment here, I'm going to assume it's the configurations bit with the resolution strategy. 

The issue isn't how to add spring boot at the correct level to test, which you are correct that adding `testImplementation` would do. It's how to correctly _remove_ the deps introduced by the `compile` declarations that persist into the test phase. (Possibly `compileOnly` could address that, but I didn't want to go there, because I didn't want to alter the way the actual artifact was built to avoid the pom ending up with its dependencies missing.) The configurations block subs in the test versions for the compile versions, giving the best of both worlds.